### ISSUE TITLE
DSL.currentTimestamp() should generate GETDATE() instead of CURRENT_BIGDATETIME() in Sybase ASE 12.5

### ISSUE THREAD

**Issue by luckfamousa** [jOOQ/jOOQ#10365]
### Expected behavior and actual behavior:

Actual: Using `DSL.currentLocalDateTime` in a query against an older Sybase ASE results in a DataAccessException with message `Function 'current_bigdatetime' not found.`

Expected: `DSL.currentLocalDateTime` to map on a vendor specific function that returns the current local datetime. Probably `GetDate()`.

### Steps to reproduce the problem:

Any query against an older Sybase ASE that uses `DSL.currentLocalDateTime` .

### Versions:

- jOOQ: 3.12.3
- Java: 13
- Database (include vendor): Sybase ASE 12.5.1.516
- OS: Any
- JDBC Driver (include name if inofficial driver): jConnect (TM) for JDBC(TM)/6.05(Build 25904)/P/EBF13251/JDK14/Wed Mar  8 11:04:35 2006

**Comment by lukaseder** [comment]
Thank you very much for your report. I will look into this shortly.

**Comment by lukaseder** [comment]
Thank you very much for your patience. The `current_bigdatetime()` function seems to have been made available in ASE 15.5:
http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36271.1550/html/blocks/CJABCHJJ.htm

ASE 12.5.x doesn't have it yet:
http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.dc36271_1251/html/blocks/CHDIHGDF.htm

We should introduce versioned dialects to be able to distinguish between the versions. From what I can tell, in later versions, using `current_bigdatetime()` is preferrable, because the precision is higher, e.g.:

```sql
select 
  datepart(ms, a), 
  datepart(us, a), 
  datepart(ms, b),
  datepart(us, b)
from (select getdate() a, current_bigdatetime() b) t
```

Yields:

```
   | |   |      |
---|-|---|------|
233|0|235|235297|
```

As a workaround, in the meantime, I recommend using plain SQL templating:
https://www.jooq.org/doc/latest/manual/sql-building/plain-sql-templating/

I.e. 

```java
public static Field<LocalDateTime> getdate() {
    return DSL.field("getdate()", SQLDataType.LOCALDATETIME);
}
```

**Comment by lukaseder** [comment]
This depends on versioned dialects for the `SQLDialect.ASE` family: https://github.com/jOOQ/jOOQ/issues/10404

**Comment by luckfamousa** [comment]
Yes, versioned dialects would IMO be the way to go.

Exactly your workaround is what we are using right now.

**Comment by lukaseder** [comment]
This is fixed for 3.14.0 and will be included in the next nightly build. 3.14.0-SNAPSHOT downloads are available from here: https://www.jooq.org/download/versions

As it depends on a new feature (versioned dialects: #10404) with significant impact, we're not backporting it to 3.13.
